astropy/units/core.py

```diff
@@ -2099,13 +2107,31 @@ def __call__(
                 pass
 
             try:
-                return f.parse(s)
+                with (
+                    _WARNING_LOCK,
+                    warnings.catch_warnings(
+                        action=_WARNING_ACTIONS[parse_strict],
+                        category=UnitParserWarning,
+                    ),
+                ):
+                    return f.parse(s)
             except NotImplementedError:
                 raise
+            except UnitParserWarning as err:
+                new_err = ValueError(err)
+                new_err.add_note(
+                    "If you cannot change the unit string then try specifying the "
+                    "'parse_strict' argument."
+                )
+                raise new_err from err
+            except KeyError as err:
+                if parse_strict in _WARNING_ACTIONS:
```

Comment:
This should be `parse_strict not in _WARNING_ACTIONS`!

Reply:
This is correct as it is and it is tested by `astropy/units/tests/test_units.py::test_parse_strict_noncritical_error["invalid"]`.

Reply:
Oops, my mistake!